internal/parser/scanner/ruleset/ruleset_default.go

```diff
@@ -186,3 +192,24 @@ func defaultUnquotedLiteralRule(s RuneScanner) (token.Type, bool) {
 
 	return token.Literal, true
 }
+
+func defaultNumericLiteralRule(s RuneScanner) (token.Type, bool) {
+	if next, ok := s.Lookahead(); !(ok && defaultNumericLiteral.Matches(next)) {
+		fmt.Println(string(next))
```

Comment:
No.
```suggestion
```

Reply:
Never.
Always use a debugger instead of printing to stdout! Prevents such things from happening

Reply:
Not a final PR, Ill clean those up before a final one :D